### PR TITLE
Move initialization tasks from OCSInit to StorageClusterInit

### LINKED ISSUE #69: OCS creates storageclasses with wrong provisioners
According to https://github.com/rook/rook/commit/9f6c0ba2c59b46ea0b71de557f41a9310ca3dc89#diff-dbe583328ce8f4478e6407fc2c616c54
the provisioner name is dynamic now and should start with the namespace in which it is deployed.
For OCS also it should be this way.
Currently, OCS creates provisioner with rook-ceph prefix, even though its deployed in openshift-storage namespace.
Screenshot from 2019-08-30 00-36-43

### PR DESCRIPTION
**Pull request by jarrpa**
* Enable `Conditions` for StorageClusterInit
* Move initialization tasks from OCSInit to StorageClusterInit
* Minor fixes

Resolves #69

### CODE REVIEW

**Comment by jarrpa** [comment]
@mhrivnak I'm getting a weird error when I try to regenerate stuff from the operator-sdk:
```console
# GO111MODULE=on build/_output/tools/operator-sdk-v0.10.0-x86_64-linux-gnu generate k8s --verbose
DEBU[0000] Debug logging is set                         
INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [ocs:[v1alpha1], ] 
F0829 08:54:41.733132    1147 deepcopy.go:885] Hit an unsupported type invalid type for invalid type, from github.com/openshift/ocs-operator/pkg/apis/ocs/v1alpha1.OCSInitialization

# GO111MODULE=on build/_output/tools/operator-sdk-v0.10.0-x86_64-linux-gnu generate openapi --verbose
DEBU[0000] Debug logging is set                         
INFO[0000] Running OpenAPI code-generation for Custom Resource group versions: [ocs:[v1alpha1], ] 
ERROR: logging before flag.Parse: E0829 08:55:28.555225    1238 openapi.go:365] Error when generating: TypeMeta, TypeMeta invalid type
Error: openapi-gen generator error: Failed executing generator: some packages had errors:
cannot generate spec for type invalid type
```

Any ideas?

**Comment by mhrivnak** [comment]
Works for me.

```
$ git reset --hard upstream/master
HEAD is now at 9f72402 Merge pull request #67 from kshlm/portable-device-sets
$ git checkout -b jose
Switched to a new branch 'jose'
$ git pull https://github.com/jarrpa/ocs-operator.git storageclusterinit
remote: Enumerating objects: 49, done.
remote: Counting objects: 100% (49/49), done.
remote: Compressing objects: 100% (13/13), done.
remote: Total 37 (delta 24), reused 37 (delta 24), pack-reused 0
Unpacking objects: 100% (37/37), done.
From https://github.com/jarrpa/ocs-operator
 * branch            storageclusterinit -> FETCH_HEAD
Removing pkg/controller/ocsinitialization/generate.go
Merge made by the 'recursive' strategy.
 deploy/crds/ocs_v1alpha1_ocsinitialization_crd.yaml                                    |   8 ---
 deploy/crds/ocs_v1alpha1_storageclusterinitialization_crd.yaml                         |  43 ++++++++++++++
 pkg/apis/ocs/v1alpha1/ocsinitialization_types.go                                       |  10 +---
 pkg/apis/ocs/v1alpha1/storageclusterinitialization_types.go                            |  19 ++++++-
 pkg/controller/ocsinitialization/generate.go                                           |  31 ----------
 pkg/controller/ocsinitialization/ocsinitialization_controller.go                       | 358 +------------------------------------------------------------------------------------------------------------------
 pkg/controller/storageclusterinitialization/generate.go                                |  31 ++++++++++
 pkg/controller/storageclusterinitialization/storageclusterinitialization_controller.go | 444 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-
 8 files changed, 538 insertions(+), 406 deletions(-)
 delete mode 100644 pkg/controller/ocsinitialization/generate.go
 create mode 100644 pkg/controller/storageclusterinitialization/generate.go
$ GO111MODULE=on operator-sdk generate k8s --verbose
DEBU[0000] Debug logging is set                         
INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [ocs:[v1alpha1], ] 
INFO[0003] Code-generation complete.
$ operator-sdk version
operator-sdk version: v0.10.0, commit: ff80b17737a6a0aade663e4827e8af3ab5a21170

**Comment by jarrpa** [comment]
Managed to work around the issue. Ran into this: https://github.com/operator-framework/operator-sdk/issues/1854#issuecomment-525132306 Setting GOROOT was the key.

@kshlm  @ashishranjan738 PTAL

File: pkg/controller/storageclusterinitialization/storageclusterinitialization_controller.go [hunk 1]
```diff
@@ -127,5 +133,439 @@ func (r *ReconcileStorageClusterInitialization) Reconcile(request reconcile.Requ
...
+// newStorageClasses returns the StorageClass instances that should be created
+// on first run.
+func (r *ReconcileStorageClusterInitialization) newStorageClasses(initData *ocsv1alpha1.StorageClusterInitialization) ([]storagev1.StorageClass, error) {
+	persistentVolumeReclaimDelete := corev1.PersistentVolumeReclaimDelete
+	ret := []storagev1.StorageClass{
+		storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: generateNameForCephFilesystemSC(initData),
+			},
+			Provisioner:   "rook-ceph.cephfs.csi.ceph.com",
```
Comment (umangachapagain) [inline pkg/controller/storageclusterinitialization/storageclusterinitialization_controller.go]: Provisioner name should be **storageClusterNamespace**.cephfs.csi.ceph.com

File: pkg/controller/storageclusterinitialization/storageclusterinitialization_controller.go [hunk 2]
```diff
@@ -127,5 +133,439 @@ func (r *ReconcileStorageClusterInitialization) Reconcile(request reconcile.Requ
...
+				"csi.storage.k8s.io/provisioner-secret-namespace": initData.Namespace,
+				"csi.storage.k8s.io/node-stage-secret-name":       "rook-ceph-csi",
+				"csi.storage.k8s.io/node-stage-secret-namespace":  initData.Namespace,
+			},
+		},
+		storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: generateNameForCephBlockPoolSC(initData),
+			},
+			Provisioner:   "rook-ceph.rbd.csi.ceph.com",
```
Comment (umangachapagain) [inline pkg/controller/storageclusterinitialization/storageclusterinitialization_controller.go]: Provisioner name should be **storageClusterNamespace**.cephfs.csi.ceph.com

**Comment by jarrpa** [comment]
@umangachapagain Changes applied and updated.

**Comment by obnoxxx** [comment]
@davidvossel 
> run `make gen-latest-csv` to pick up the crd changes into the bundle.

is that change request to this PR?

**Comment by davidvossel** [comment]
fyi, I have PRs open to handle making sure CI fails if the CSV is out of date.

ocs-operator side.
https://github.com/openshift/ocs-operator/pull/74/

corresponding prow release config changes
https://github.com/openshift/release/pull/4893

Once we get those merged, CI will tell us `make gen-latest-csv` needs to run before allowing us to commit changes.

**Pull request merged**